### DISCUSSION TITLE
Is there a reason for the parameter of configure function in configurable trait is not restricted to implement SetOpt?

### DISCUSSION THREAD

**Discussion by ta3pks** [sagebind/isahc#152] [Questions] [answered]
I am preparing a pr for the curl interface option. While writing the code I noticed that the options parameter does not have to implement SetOpt. Is this a mistake or am I missing something since all options Must provide a curl handle through SetOpt?

**Comment by sagebind** [comment]
`SetOpt` is a private trait that should not be exposed in the public docs, and `configure` strictly doesn't really need the parameter to implement it.

I'm considering making `configure` private as well using the sealed trait pattern before the next release. If that happens, then adding `SetOpt` wouldn't be as big of a deal.

**Comment by sagebind** [comment, marked as answer]
Nevermind, it looks like I already did set up a separate trait! `master` still has some cleanup to do yet before being released as 0.9 so there's probably a few incomplete things here and there. Requiring `SetOpt` should now make sense so I'll add that now.
